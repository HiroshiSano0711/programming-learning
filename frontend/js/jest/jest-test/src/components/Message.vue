<template>
  <li class="message">{{ message }}</li>
</template>

<script>
export default {
  props: {
    message: {
      type: String,
      required: true,
      validator: message => message.length > 1
    }
  }
};
</script>
