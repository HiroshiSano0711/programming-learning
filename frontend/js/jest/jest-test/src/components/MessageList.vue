<!-- Comment
<template>
  <ul>
    <li v-for="message in messages">{{ message }}</li>
  </ul>
</template>

<script>
export default {
  name: "list",
  props: ["messages"]
};
</script>
 -->

<template>
  <ul>
    <Message
      v-for="message in messages"
      :message="message"
    />
  </ul>
</template>

<script>
import Message from './Message'

export default {
  components: {
    Message
  },
  props: ['messages']
}
</script>
