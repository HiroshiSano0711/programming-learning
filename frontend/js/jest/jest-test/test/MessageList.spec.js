// import { mount } from '@vue/test-utils';
// import MessageList from '../src/components/MessageList';

// describe('MessageList.test.js', () => {
//   let cmp;

//   beforeEach(() => {
//     cmp = mount(MessageList, {
//       propsData: { messages: ['Cat'] }
//     });
//   });

//   it('has received Cat as the message property', () => {
//     expect(cmp.vm.messages).toEqual(['Cat'])
//   });

//   it('has the expected html structure', () => {
//     expect(cmp.element).toMatchSnapshot()
//   });
// });

// import { shallowMount } from "@vue/test-utils";
// import MessageList from "../src/components/MessageList";

// describe("MessageList.test.js", () => {
//   let cmp;

//   beforeEach(() => {
//     cmp = shallowMount(MessageList, { // Create a shallow instance of the component
//       propsData: {
//         messages: ["Cat"]
//       }
//     });
//   });

//   it('equals messages to ["Cat"]', () => { // Within cmp.vm, we can access all Vue instance methods
//     expect(cmp.vm.messages).toEqual(["Cat"]);
//   });

//   it("has the expected html structure", () => {
//     expect(cmp.element).toMatchSnapshot();
//   });
// });

import { mount } from "@vue/test-utils";
import MessageList from "../src/components/MessageList";
import Message from "../src/components/Message";

describe("MessageList.test.js", () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(MessageList, {
      propsData: {
        messages: ["Cat"]
      }
    });
  });

  it('messageプロパティとして["Cat"]を受け取ること', () => {
    expect(cmp.vm.messages).toEqual(["Cat"]);
  });

  it("期待されるHTML構造であること", () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it("MessageListコンポーネントであること", () => {
    expect(cmp.is(MessageList)).toBe(true);
  });
  
  it("Messageコンポーネントが含まれること", () => {
    expect(cmp.findComponent(Message).exists()).toBe(true);
  });

  it("MessageListとMessageの両方はVueインスタンスである", () => {
    expect(cmp.vm).toBeTruthy();
    expect(cmp.findComponent(Message).vm).toBeTruthy();
  });

  it("message要素が存在すること", () => {
    expect(cmp.find(".message").exists()).toBe(true);
  });

  it("Messageコンポーネントは存在しないこと", () => {
    expect(cmp.exists(Message)).toBe(true);
  });

  it('Messageは"message"のクラス属性をセットすること', () => {
    expect(cmp.findComponent(Message).attributes().class).toBe("message");
  });
});
