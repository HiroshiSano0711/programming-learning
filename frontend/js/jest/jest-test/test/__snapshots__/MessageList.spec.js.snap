// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.test.js has the expected html structure 1`] = `
<ul>
  <li
    class="message"
  >
    Cat
  </li>
</ul>
`;

exports[`MessageList.test.js 期待されるHTML構造であること 1`] = `
<ul>
  <li
    class="message"
  >
    Cat
  </li>
</ul>
`;
